### PR TITLE
Drop Julia 0.6

### PR DESCRIPTION
**Pull request by mjram0s**

### CODE REVIEW

File: src/LayerDicts.jl [hunk 1]
```diff
@@ -1,8 +1,6 @@
 __precompile__()
```
Comment (oxinabox) [inline src/LayerDicts.jl]:
```suggestion
```

File: src/LayerDicts.jl [hunk 2]
```diff
@@ -66,37 +64,19 @@ end
...
 Base.length(ld::LayerDict) = length(keys(ld))
 
-if VERSION < v"0.7.0-DEV.5126"
-    function Base.start(ld::LayerDict)
-        ld_keys = keys(ld)
-        return (ld_keys, start(ld_keys))
-    end
+function Base.iterate(ld::LayerDict, state=(keys(ld),))
+    ld_keys = first(state)
+    key_state = Base.tail(state)
```
Comment (oxinabox) [inline src/LayerDicts.jl:68]:
```suggestion
    key_state = Base.tail(state)
    @assert length(key_state) == 0 || length(key_state) == 1
```

Is my suggestion above correct? if it is not, then then this code is really confusing.

File: src/LayerDicts.jl [hunk 3]
```diff
@@ -66,37 +64,19 @@ end
...
-    end
+    iter = iterate(ld_keys, key_state...)
 
-    function Base.next(ld::LayerDict, state)
-        ld_keys, key_state = state
-        key, new_key_state = next(ld_keys, key_state)
-        return (key => ld[key]), (ld_keys, new_key_state)
+    if iter === nothing
+        return nothing
     end
```
Comment (oxinabox) [inline src/LayerDicts.jl]: I would shorten the  above `if` into  `iter === nothing && return nothing`.
But I guess that is just personal preference, so do not feel compelled to make this change if you think the `if` is cleaner.

File: src/LayerDicts.jl (hunk 2 above)
Comment (rofinn) [reply src/LayerDicts.jl:68]: That seems reasonable to me, though I'm not super familiar with the new iteration protocol.

**Review by oxinabox — APPROVED** [review]

**Pull request merged**